### PR TITLE
Fixed undefined array key category_name in statistics

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Fixed undefined array key category_name in statistics 
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Sponsor company   | PrestaShop SA 
| How to test?      | See https://github.com/PrestaShop/blockwishlist/pull/276#pullrequestreview-2275891782

### CODE REVIEW

File: src/Calculator/StatisticsCalculator.php [hunk 1]
```diff
@@ -189,18 +183,31 @@ public function getProductImage($productDetails)
...
+    {
+        $imgDetails = [];
 
         foreach ($presentedProduct as $key => $value) {
             if ($key == 'embedded_attributes') {
                 $imgDetails = $value['cover'];
             }
         }
         if (!$imgDetails) {
+            $imageRetriever = new ImageRetriever($this->context->link);
```
Comment (nicosomb) [inline src/Calculator/StatisticsCalculator.php:210]: I'm not fan of this line because I already have it line 150.

**Review by tleon — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by paulnoelcholot** [comment]
Hello @nicosomb,

I tested your PR and I found bug sorry :/

Thanks!

https://www.loom.com/share/7eb1aa91d2f444c0800dca95500cc5c3?sid=6f4ffc33-28d5-4a25-949b-575c4531e88a

**Comment by Hlavtox** [comment]
@paulnoelcholot I think you must test on 8.0.x or 8.1.x to reproduce this.

**Comment by nicosomb** [comment]
@paulnoelcholot you already found this bug in an other pull request, there is an opened issue for that : https://github.com/PrestaShop/PrestaShop/issues/33374

**Comment by Hlavtox** [comment]
Bump @PrestaShop/qa-functional please, I need this released so https://github.com/PrestaShop/PrestaShop/pull/36873 can move on. :-)

**Comment by boherm** [comment]
Hi @nicosomb! I have just test your PR with https://github.com/PrestaShop/PrestaShop/pull/36880 and it's ok!

**Review by boherm — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**